numba/targets/builtins.py

```diff
@@ -49,6 +47,36 @@ def generic_is(context, builder, sig, args):
     else:
         return cgutils.false_bit
 
+
+@lower_builtin('==', types.Any, types.Any)
+def generic_eq(context, builder, sig, args):
+    """
+    Default implementation for `x == y` that fallback to `is`
+    """
+    lhs_type, rhs_type = sig.args
+    if isinstance(lhs_type, types.UserNe) or isinstance(rhs_type, types.UserNe):
+        warnings.warn("Possible unintentional usage of default __eq__ on "
+                      "object that implements __ne__", UserWarning)
+
+    is_impl = context.get_function("is", sig)
+    out = is_impl(builder, args)
+    return impl_ret_new_ref(context, builder, sig.return_type, out)
+
+
+@lower_builtin('!=', types.Any, types.Any)
+def generic_ne(context, builder, sig, args):
+    """
+    Default implementation for `x != y` that fallback to `is not`
```

Comment:
I'm not sure I understand. Your comment on the PR indicates that `x != y` should fallback on `not (x == y)`.